### ISSUE TITLE
UnicodeEncodeError: 'ascii' codec can't encode character u'\u258f' in position 20: ordinal not in range(128)

### ISSUE THREAD

**Issue by davidlandry93** [neishm/fstd2nc#19]
The conversion fails when the environment variable LC_ALL is not set.

```
dav000@eccc1-ppp2 [17:35:13] ~/site1/data $ fstd2nc $CMCGRIDF/prog/regeta/2019082700_012 test2.nc
c_fstopi option MSGLVL set to 4
Found 1 RPN input file(s)
Warning: Multiple sfclevel axes.  Appending integer suffixes to their names.
Warning: Multiple level axes.  Appending integer suffixes to their names.
Warning: Renaming '5P' to '_5P'.

Traceback (most recent call last):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.0_all/bin/fstd2nc", line 9, in <module>
    load_entry_point('fstd2nc==0.20180821.0', 'console_scripts', 'fstd2nc')()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.0_all/lib/python/fstd2nc/__main__.py", line 105, in _fstd2nc_cmdline_trapped
    _fstd2nc_cmdline (*args, **kwargs)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.0_all/lib/python/fstd2nc/__main__.py", line 99, in _fstd2nc_cmdline
    buf.to_netcdf(outfile, nc_format=nc_format, global_metadata=global_metadata, zlib=zlib, progress=progress)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.0_all/lib/python/fstd2nc/mixins/netcdf.py", line 376, in to_netcdf
    for r,shape,v,ind in bar.iter(sorted(io)):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/__init__.py", line 125, in iter
    self.next()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/__init__.py", line 73, in next
    self.update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.0_all/lib/python/fstd2nc/mixins/__init__.py", line 173, in update
    super(_ProgressBar,self).update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/bar.py", line 86, in update
    self.writeln(line)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/helpers.py", line 68, in writeln
    print(line, end='', file=self.file)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u258f' in position 20: ordinal not in range(128)
```

The following command works, however:
```
LC_ALL=en_US.utf-8 fstd2nc $CMCGRIDF/prog/regeta/2019082700_012 test3.nc
```

Is there a way the converter could be less sensitive to this environment variable?
Thanks!

**Comment by davidlandry93** [comment]
Hmm it still fails on my end:

``` sh
dav000@eccc1-ppp1 [14:14:11] ~/site1/data $ . ssmuse-sh -x eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2
Loading dependencies
. r.load.dot eccc/mrd/rpn/MIG/ENV/py/2.7/rpnpy/2.1.b2
shortcut.dot eccc/mrd/rpn/MIG/ENV/py/2.7/rpnpy/2.1.b2
INFO: using delayed mode
USING DOMAIN /fs/ssm/eccc/mrd/rpn/MIG/ENV/d/py/2.7/rpnpy/rpnpy_2.1.b2
. /fs/ssm/eccc/mrd/rpn/MIG/ENV/d/py/2.7/rpnpy/rpnpy_2.1.b2/rpnpy_2.1.b2_all/.setenv.__extlib__.ubuntu-14.04-amd64-64.dot
. /fs/ssm/eccc/mrd/rpn/MIG/ENV/d/py/2.7/rpnpy/rpnpy_2.1.b2/rpnpy_2.1.b2_all/.setenv.defaults.dot
. ssmuse-sh -p eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625
dav000@eccc1-ppp1 [14:14:13] ~/site1/data $ fstd2nc $CMCGRIDF/prog/regeta/2019082700_012 test5.nc
c_fstopi option MSGLVL set to 4
Found 1 RPN input file(s)
Warning: Multiple sfclevel axes.  Appending integer suffixes to their names.
Warning: Multiple level axes.  Appending integer suffixes to their names.
Warning: Renaming '5P' to '_5P'.

Traceback (most recent call last):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2_all/bin/fstd2nc", line 9, in <module>
    load_entry_point('fstd2nc==0.20180821.1', 'console_scripts', 'fstd2nc')()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2_all/lib/python/fstd2nc/__main__.py", line 105, in _fstd2nc_cmdline_trapped
    _fstd2nc_cmdline (*args, **kwargs)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2_all/lib/python/fstd2nc/__main__.py", line 99, in _fstd2nc_cmdline
    buf.to_netcdf(outfile, nc_format=nc_format, global_metadata=global_metadata, zlib=zlib, progress=progress)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2_all/lib/python/fstd2nc/mixins/netcdf.py", line 376, in to_netcdf
    for r,shape,v,ind in bar.iter(sorted(io)):
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/__init__.py", line 125, in iter
    self.next()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/__init__.py", line 73, in next
    self.update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20180821.2_all/lib/python/fstd2nc/mixins/__init__.py", line 178, in update
    super(_ProgressBar,self).update()
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/bar.py", line 86, in update
    self.writeln(line)
  File "/fs/ssm/eccc/crd/ccmr/EC-CAS/master/fstd2nc-deps_20180625_ubuntu-14.04-amd64-64/lib/python/progress/helpers.py", line 68, in writeln
    print(line, end='', file=self.file)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u258f' in position 20: ordinal not in range(128)
```

**Comment by neishm** [comment]
Hi @davidlandry93, could you paste the result of running `locale` in your environment?

**Comment by davidlandry93** [comment]
Here. It's probably not normal to have nothing defined in there -_- But who knows what happens once you boot ordenv
```
LANG=C.UTF-8
LANGUAGE=
LC_CTYPE="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_COLLATE="C"
LC_MONETARY="C"
LC_MESSAGES="C"
LC_PAPER="C"
LC_NAME="C"
LC_ADDRESS="C"
LC_TELEPHONE="C"
LC_MEASUREMENT="C"
LC_IDENTIFICATION="C"
LC_ALL=C
```

**Comment by neishm** [comment]
@davidlandry93 you have a very interesting environment.
I just updated the ssm domain in-place with another potential fix.  Does it still have the same issue for you?

**Comment by davidlandry93** [comment]
That fixed it! thanks

**Comment by neishm** [comment]
I have a new release available for testing now (version 0.20190903.0) which includes this fix along with the new `--compression` option (#20).

Internally, can be loaded by running
```
. ssmuse-sh -p eccc/crd/ccmr/EC-CAS/master/fstd2nc_0.20190903.0
```
Should also be available externally via `pip` by running
```
pip install --upgrade fstd2nc progress
```
